Net/SMTP.php

```diff
@@ -623,23 +623,43 @@ public function starttls()
                     /* STREAM_CRYPTO_METHOD_TLS_ANY_CLIENT constant does not exist
                      * and STREAM_CRYPTO_METHOD_SSLv23_CLIENT constant is
                      * inconsistent across PHP versions. */
-                    $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT
-                    | @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT
-                    | @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
+                    $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT;
+
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT;
+                    }
+
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
+                    }
+
+                    //New Crypto-Method since PHP7.4
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT;
+                    }
                 }
-                if (PEAR::isError($result = $this->socket->enableCrypto(true, $crypto_method))) {
+
+                $result = 0;
+                while ($result == 0) {
+                    $result = $this->socket->enableCrypto(true, $crypto_method);
+                    if ($this->socket->isBlocking() !== true) {
+                        usleep(1);
+                    }
+                }
```

Comment:
This _could_ loop indefinitely. Let's limit the number of attempts.

Reply:
This could also be expressed as a `do .. while` loop. It always makes at least one iteratin.

Reply:
@jparise thank you. good response. i pushed another commit with limiting the number of loop attempts.